src/battle_dome.c

```diff
@@ -4130,6 +3928,244 @@ static u8 Task_GetInfoCardInput(u8 taskId)
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeLuckyMove(u32 move)
+{
+    if (gBattleMoves[move].accuracy <= 50)
+        return TRUE;
+    switch(gBattleMoves[move].effect)
+    {
+    case EFFECT_COUNTER:
+    case EFFECT_OHKO: // Technically redundant because of the above accuracy check
+    case EFFECT_METRONOME:
+    case EFFECT_MIRROR_MOVE:
+    case EFFECT_SKETCH:
+    case EFFECT_SLEEP_TALK:
+    case EFFECT_PRESENT:
+    case EFFECT_ASSIST:
+    case EFFECT_MAGIC_COAT:
+    case EFFECT_REVENGE:
+    case EFFECT_IMPRISON:
+    case EFFECT_SNATCH:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomePopularMove(u32 move)
+{
+    u8 i;
+    for (i = 0; i < NUM_TECHNICAL_MACHINES + NUM_HIDDEN_MACHINES; i++)
+    {
+        if (ItemIdToBattleMoveId(ITEM_TM01 + i) == move)
+            return TRUE;
+    }
+    if (i == NUM_TECHNICAL_MACHINES + NUM_HIDDEN_MACHINES)
+        return FALSE;
+    // Filter in TMs/HMs
+    if (gBattleMoves[move].power >= 90)
+        return TRUE;
+
+    switch(gBattleMoves[move].effect)
+    {
+    case EFFECT_PROTECT:
+    case EFFECT_MAT_BLOCK:
+    case EFFECT_ATTACK_UP_2:
+    case EFFECT_SPECIAL_ATTACK_UP_2:
+    case EFFECT_SPECIAL_ATTACK_UP_3:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeStatusMoveEffect(u32 effect)
+{
+    switch(effect)
+    {
+    case EFFECT_TRAP:
+    case EFFECT_SLEEP:
+    case EFFECT_CONFUSE:
+    case EFFECT_DISABLE:
+    case EFFECT_POISON:
+    case EFFECT_PARALYZE:
+    case EFFECT_TOXIC:
+    case EFFECT_LEECH_SEED:
+    case EFFECT_TAUNT:
+    case EFFECT_TORMENT:
+    case EFFECT_WILL_O_WISP:
+    case EFFECT_ENCORE:
+    case EFFECT_ATTRACT:
+    case EFFECT_NIGHTMARE:
+    case EFFECT_YAWN:
+    case EFFECT_CURSE:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeRareMove(u32 move)
+{
+    u16 i, j;
+    u16 species = 0;
+    for(i = 0; i < NUM_SPECIES; i++)
+    {
+        for(j = 0; gLevelUpLearnsets[i][j].move != LEVEL_UP_MOVE_END; j++)
+        {
+            if (gLevelUpLearnsets[i][j].move == move)
+            {
+                species++;
+                break;
+            }
+        }
+        if (species > NUM_SPECIES / 10)
+            return FALSE;
+    }
+    return TRUE;
+}
+
+static bool32 IsDomeComboMoveEffect(u32 effect)
+{
+    switch(effect)
+    {
+    // Weather moves
+    case EFFECT_SUNNY_DAY:
+    case EFFECT_RAIN_DANCE:
+    case EFFECT_SANDSTORM:
+    case EFFECT_HAIL:
+    case EFFECT_SNOWSCAPE:
+    // Terrain moves
+    case EFFECT_GRASSY_TERRAIN:
+    case EFFECT_ELECTRIC_TERRAIN:
+    case EFFECT_MISTY_TERRAIN:
+    case EFFECT_PSYCHIC_TERRAIN:
+    // Moves dependent on weather
+    case EFFECT_SYNTHESIS:
+    case EFFECT_MORNING_SUN:
+    case EFFECT_MOONLIGHT:
+    case EFFECT_SHORE_UP:
+    case EFFECT_THUNDER:
+    case EFFECT_HURRICANE:
+    //case EFFECT_BLIZZARD:
```

Comment:
Because they don't have actual unique effects and the move ID is checked instead in the battle code. I'm not sure if this means we should just give them their own effect so that we can easily check for their uniqueness outside of the battle scripts too (which would maybe be out of scope for this PR?) or if we should just quietly forget about them, haha

Reply:
Yeah, it'd be out of scope for this PR I feel. Leave a note for now :)